messages/helpers.go

```diff
@@ -183,6 +183,24 @@ func AllHaveLowerRound(messages []*proto.Message, round uint64) bool {
 	return true
 }
 
+// AllHaveSameRound checks if all messages have the same round
+func AllHaveSameRound(messages []*proto.Message) bool {
+	if len(messages) <= 1 {
+		return false
+	}
+
+	var round = messages[0].View.Round
+
+	for _, message := range messages[1:] {
```

Comment:
I went ahead and benchmarked this, to see how much we would save on this single element iteration (you're not including the first element in this set). It turns out this method is around ~20% without iterating over the first element. 
Here's [the snippet](https://goplay.tools/snippet/5RIZYNdWwvy) I used.

You can use this same idea with `AllHaveSameHeight`

Reply:
Plays nice with <= 1 comment so fixed as well.